### PR TITLE
RHTAPSRE-439: Deploy CA bundle configmap

### PR DESCRIPTION
**Pull request by gbenhaim**
Openshift will inject the CA bundle to the configmap. The configmap will be available in every tenant namespace, so tenant's pipelines would able to reach services that use private pki.

 https://docs.openshift.com/container-platform/4.14/networking/configuring-a-custom-pki.html#nw-proxy-configure-object_configuring-a-custom-pki

### CODE REVIEW

File: deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml [hunk 1]
```diff
@@ -262,6 +262,13 @@ objects:
   - kind: ServiceAccount
     namespace: ${SPACE_NAME}-tenant
     name: appstudio-pipeline
+- apiVersion: v1
+  data: {}
+  kind: ConfigMap
+  metadata:
+    labels:
+      config.openshift.io/inject-trusted-cabundle: "true"
+    name: trusted-ca
```
Comment (mfrancisc) [inline deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml:271]: should we add the namespace ? 
```suggestion
    name: trusted-ca
    namespace: ${SPACE_NAME}-tenant
```

Comment (gbenhaim) [reply deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml:271]: added. thanks!

**Review by mfrancisc — APPROVED** [review]
LGTM

**Comment by gbenhaim** [comment]
@mfrancisc is it an issue with my change or an infra issue?

**Comment by mfrancisc** [comment]
/retest

potential flaky https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/codeready-toolchain_host-operator/978/pull-ci-codeready-toolchain-host-operator-master-e2e/1764977752739418112/build-log.txt

**Comment by mfrancisc** [comment]
/retest

some strange error with the host-operator Subscription:
```
  conditions:
  - lastTransitionTime: "2024-03-05T15:47:22Z"
    message: all available catalogsources are healthy
    reason: AllCatalogSourcesHealthy
    status: "False"
    type: CatalogSourcesUnhealthy
  - message: 'failed to populate resolver cache from source source-toolchain-host-operator-latest/toolchain-host-05154206:
      failed to list bundles: rpc error: code = Unavailable desc = connection error:
      desc = "transport: Error while dialing dial tcp: lookup source-toolchain-host-operator-latest.toolchain-host-05154206.svc
      on 172.30.0.10:53: no such host"'
    reason: ErrorPreventedResolution
    status: "True"
    type: ResolutionFailed
  lastUpdated: "2024-03-05T15:47:22Z"
```

see: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/codeready-toolchain_host-operator/978/pull-ci-codeready-toolchain-host-operator-master-e2e/1765031212323180544/build-log.txt

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**